app/models/main.py

```diff
@@ -635,6 +622,32 @@ def serialize_agreed_variation(agreed_variation, with_users=False):
             "agreedUserEmail": user.email_address,
         })
 
+    @property
+    def application_status(self):
+        submitted_services = DraftService.query.filter(
+            DraftService.status.in_(('submitted', 'failed',)),
+            DraftService.framework_id == self.framework_id,
+            DraftService.supplier_id == self.supplier_id
+        ).count()
```

Comment:
We might not actually need to run this query at all depending on whether company details are required, so some re-ordering of the logic here might make things much more efficient in certain cases.